spring-integration-sftp/src/main/java/org/springframework/integration/sftp/gateway/SftpOutboundGateway.java

```diff
@@ -129,4 +133,26 @@ public String getComponentType() {
 		return "sftp:outbound-gateway";
 	}
 
+	@Override
+	public boolean isChmodCapable() {
+		return true;
+	}
+
+	@Override
+	protected void doChmod(RemoteFileTemplate<LsEntry> remoteFileTemplate, String path, int chmod) {
+		remoteFileTemplate.executeWithClient(new ClientCallbackWithoutResult<ChannelSftp>() {
+
+			@Override
+			protected void doWithClientWithoutResult(ChannelSftp client) {
+				try {
+					client.chmod(chmod, path);
```

Comment:
`path` and `chmod` must be `final` in the method definition.
See Travis failure.